### PR TITLE
Updating the Nightwatch axeCheckJS helper method for e2e tests

### PR DESCRIPTION
**Pull request by 1Copenut**
## Description
During a manual review, we noticed a few errors were being picked up by the axe-core CLI and browser plugin but not our axeCheckJS tests run out of Nightwatch.

* This work previously fixed NW tests for e2e and static content. The static content fix has been moved to `content-build` in draft PR https://github.com/department-of-veterans-affairs/content-build/pull/151
* Removed `axeCheck()` middleware for filtering out color contrast and accessible link name
* ~~Color contrast issues appear to be resolved in axe-core 4.1.1~~
* Accessible link name is a WCAG2.1 success criterion and can be addressed in a follow on release

## Testing done
* Ran the test suite locally using `$ yarn test:e2e:headless` before and after removing middleware from axeCheckJS
* Verified issues were being logged to console and test suite exited with `CODE 1` after removing middleware
* Verified the Nighwatch e2e tests are performing as expected and passing in Jenkins CI

## Screenshots
- [ ] See https://github.com/department-of-veterans-affairs/va.gov-team/issues/21738

## Acceptance criteria
- [ ] See https://github.com/department-of-veterans-affairs/va.gov-team/issues/21738

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by timwright12 — APPROVED** [review]
Tested and looks good.

File: src/platform/site-wide/tests/sitemap/sitemap-helpers.js [hunk 1]
```diff
@@ -64,7 +64,15 @@ function runTests(client, segment, only508List) {
       .waitForElementVisible('body', Timeouts.normal)
       .axeCheck(
         'document',
-        only508 ? { scope: url, rules: ['section508'] } : { scope: url },
+        only508
```
Comment (1Copenut) [inline src/platform/site-wide/tests/sitemap/sitemap-helpers.js:67]: If we're deleting all of the sitemap tests, do we still need this `sitemap-helpers.js` file?

Comment (timwright12) [reply src/platform/site-wide/tests/sitemap/sitemap-helpers.js:67]: we do because the virutal regress tests apparently use the helpers in there

**Review by kelsonic — APPROVED** [review]

**Review by Mottie — APPROVED** [review]
ghe-emoji

**Comment by timwright12** [comment]
🥳

**Pull request merged**